File: 08. Analisis Kompleksitas/C. Bilangan Agak Prima/c_bilangan_agak_prima.py
N = int(input())
for i in range(N):
    num = int(input())
    factor = 0
    j = 2
    while j*j <= num:
        if num % j == 0:
            factor += 1
        if factor > 1:
            break
        j += 1
    if (factor <= 1):
        print("YA")
    else:
        print("BUKAN")

File: 08. Analisis Kompleksitas/C. Bilangan Agak Prima/c_bilangan_agak_prima.rs
fn main() {
    let mut reader = String::new();
    std::io::stdin().read_line(&mut reader).expect("");
    let n: i32 = reader.trim().parse().expect("");
    for _ in 0..n {
        let mut reader = String::new();
        std::io::stdin().read_line(&mut reader).expect("");
        let num: i32 = reader.trim().parse().expect("");
        let mut factor: i32 = 0;
        for j in 1..num + 1 {
            if num % j == 0 {
                factor += 1;
            }
            if factor > 4 {
                break;
            }
        }
        if factor <= 4 {
            println!("YA");
        } else {
            println!("BUKAN");
        }
    }
}
